uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -386,6 +424,10 @@ define([
        * @returns {Promise}
        */
       function initDayTypesAndStatus() {
+        if (vm.leaveRequestDayTypes.length && vm.leaveRequestStatuses.length) {
```

Comment:
Why are you doing this? This function seem to be called only once. Also, isn't the api request cached anyway?